hyperspy/_signals/complex_signal.py

```diff
@@ -260,6 +261,59 @@ def angle(self, deg=False):
         return super().angle(angle, deg=deg)
     angle.__doc__ = ComplexSignal_mixin.angle.__doc__
 
+    def argand_diagram(self, size=[256, 256], display_range=None):
+        """
+        Calculate and plot Argand diagram of complex signal
+
+        Parameters
+        ----------
+        size : [int, int], optional
+            Size of the Argand plot in pixels
+            (Default: [256, 256])
+        display_range : array_like, shape(2,2) or shape(2,) optional
+            The position of the edges of the diagram
+            (if not specified explicitly in the bins parameters): [[xmin, xmax], [ymin, ymax]].
+            All values outside of this range will be considered outliers and not tallied in the histogram.
+            (Default: None)
+
+        Returns
+        -------
+        sap:
+            Argand diagram plot as Signal2D
+
+        Examples
+        --------
+        >>> import hyperspy.api as hs
+        >>> holo = hs.datasets.example_signals.object_hologram()
+        >>> ref = hs.datasets.example_signals.reference_hologram()
+        >>> w = holo.reconstruct_phase(ref)
+        >>> w.argand_diagram(display_range=[-3, 3]).plot()
+
+        """
+        im = self.imag.data.ravel()
+        re = self.real.data.ravel()
+
+        if display_range:
+            if np.asarray(display_range).shape == (2, ):
+                display_range = [[display_range[0], display_range[1]],
+                                 [display_range[0], display_range[1]]]
+            elif np.asarray(display_range).shape != (2, 2):
+                raise ValueError('display_range should be array_like, shape(2,2) or shape(2,).')
+
+        ap = np.histogram2d(re, im, bins=size, range=display_range)
```

Comment:
I'd recommend turning this into `ap, xedges, yedges = np.histogram2d(re, im, bins=size, range=display_range)` and then in the next line use `sap = Signal2D(ap.T)`. That makes it clearer that the object in `ap[0]` isn't of the same kind as `ap[1]` etc. What does `ap` stand for? You might want to give it a better name!

Reply:
`real_axis_edges` and `imag_axis_edges` (or something along these line) would be clearer than `xedges` and `yedges`, which are a bit confusing since once can be mislead with `x` and `y` referring to the x, y axes of the original image.